src/main/java/com/scalyr/integrations/kafka/AddEventsClient.java

```diff
@@ -215,47 +221,53 @@ private AddEventsResponse addEventsWithRetry(byte[] addEventsPayload, long uncom
         byte[] decompressedPayload = getDecompressedPayload(addEventsPayload);
         payloadLogger.error("Add events too large payload: {}", new String(decompressedPayload));
       }
-      return PAYLOAD_TOO_LARGE;
+      addEventsResult.complete(PAYLOAD_TOO_LARGE);
+      return;
     }
 
-    AddEventsResponse addEventsResponse = TIMEOUT_ADD_EVENTS_RESPONSE;
-    long delayTimeMs = initialBackoffDelayMs;
-    httpPost.setEntity(new ByteArrayEntity(addEventsPayload));
+    // Retries timed out
+    if (remainingMs(startTimeMs) == 0) {
```

Comment:
(just thhinking out loud) - I was wondering if ``remainingMs()`` can ever return a negative value, but I see it's guarded with max clause which will return 0 in such cases.